lib/prelude.dx

```diff
@@ -630,7 +683,7 @@ def reduce (identity:a) (combine:(a->a->a)) (xs:n=>a) : a =
 -- TODO: call this `scan` and call the current `scan` something else
 def scan' (init:a) (body:n->a->a) : n=>a = snd $ scan init \i x. dup (body i x)
 -- TODO: allow tables-via-lambda and get rid of this
-def fsum (xs:n=>Float) : Float = yieldAccum \ref. for i. ref += xs i
+def fsum (xs:n=>Float) : Float = yieldAccum (AddMonoid Float) \ref. for i. ref += xs i
 def sum  [Add v] (xs:n=>v) : v = reduce zero (+) xs
 def prod [Mul v] (xs:n=>v) : v = reduce one  (*) xs
```

Comment:
I think we can unify `fsum` and `sum` now.  Something like
```
def sum [Add a] (xs:n=>a) : a = yieldAccum (AddMonoid a) \ref. for i. ref += xs i
```
Another way would be to make `reduce` take a monoid.

Reply:
Yes, I just didn't want to fold too many changes into a single PR! Prelude cleanup is a followup I'm planning to do.